OpenRA.Mods.Common/Traits/World/MPStartLocations.cs

```diff
@@ -18,24 +18,65 @@
 
 namespace OpenRA.Mods.Common.Traits
 {
-	public class MPStartLocationsInfo : ITraitInfo
+	[Desc("Allows the map to have working spawnpoints. Also controls the 'Separate Team Spawns' checkbox in the lobby options.")]
+	public class MPStartLocationsInfo : ITraitInfo, ILobbyOptions
 	{
 		public readonly WDist InitialExploreRange = WDist.FromCells(5);
 
+		[Translate]
+		[Desc("Descriptive label for the spawn positions checkbox in the lobby.")]
+		public readonly string SeparateTeamSpawnsCheckboxLabel = "Separate Team Spawns";
+
+		[Translate]
+		[Desc("Tooltip description for the spawn positions checkbox in the lobby.")]
+		public readonly string SeparateTeamSpawnsCheckboxDescription = "Players without assigned spawn points will start as far as possible from enemy players.";
+
+		[Desc("Default value of the spawn positions checkbox in the lobby.")]
+		public readonly bool SeparateTeamSpawnsCheckboxEnabled = true;
+
+		[Desc("Prevent the spawn positions state from being changed in the lobby.")]
+		public readonly bool SeparateTeamSpawnsCheckboxLocked = false;
+
+		[Desc("Whether to display the spawn positions checkbox in the lobby.")]
+		public readonly bool SeparateTeamSpawnsCheckboxVisible = true;
+
+		[Desc("Display order for the spawn positions checkbox in the lobby.")]
+		public readonly int SeparateTeamSpawnsCheckboxDisplayOrder = 7;
```

Comment:
This should default to 0, and then override it in the individual mod rules to a position that makes sense (incrementing the order on boxes that should come later).

Reply:
As part of this, please ensure that this option always appears ahead of the Debug checkbox, and does not end up in the right hand column where the label is truncated.  e.g. for RA we will want this at 6, `LobbyPrerequisiteCheckbox@GLOBALFACTUNDEPLOY` at 7 and `LobbyPrerequisiteCheckbox@GLOBALBOUNTY` at 8.